### PR TITLE
Move standard application configuration properties into embedded appl…

### PR DESCRIPTION
**Pull request by wtritch**
…ication.yml

Modify existing dockerBuildImage gradle task to also tag built image as ":latest"
Add gradle task to push ":latest" docker image to dockerhub

This PR moves standard, base application.yml properties from the project-root application.yml to the embedded application.yml properties to support both running the service via `docker-compose` as well as via IntelliJ for command-line debugging.

Similar changes for IAT in https://github.com/SmarterApp/AP_ItemAuthoringTool/pull/679

With these changes, we can define a docker-compose entry for the service that looks like:
```
  # Item Management Service
  ap-ims:
    image: fwsbac/ap-ims:latest
    ports:
      - 8081:8080
    volumes:
      - ${HOME}:/home
    environment:
      - HOME=/home
      - SPRING_REDIS_HOST=host.docker.internal
      - SPRING_RABBITMQ_HOST=host.docker.internal
      - SPRING_DATASOURCE_URL=jdbc:postgresql://host.docker.internal:5432/iat
      - ITEMBANK_SERVICES_HISTORY_URL=http://host.docker.internal:8086
      - SECURITY_ENABLED
      - SESSION_CLUSTER_ENABLED
      - DB_USER
      - DB_PASSWORD
      - GITLAB_GROUP
      - IAT_AWS_ACCESS_KEY
      - IAT_AWS_ACCESS_SECRET
```

I'd like to do something similar for all of the IAT core services to have a single docker-compose capable of standing up the IAT ecosystem locally.

### CODE REVIEW

File: application.yml [hunk 1]
```diff
@@ -15,79 +15,6 @@
 server:
   port: 8081
```
Comment (wtritch) [inline application.yml]: Ideally this whole application.yml would be removed, and setting the server port would be via runtime-configuration environment variables in IntelliJ like:
image

File: build.gradle [hunk 2]
```diff
@@ -244,14 +244,20 @@ task dockerCopyFiles(type: Copy) {
 task dockerBuildImage(type: DockerBuildImage) {
     dependsOn 'dockerCopyFiles'
     inputDir = project.file('build/docker/')
-    tag = "${project.dockerTagBase}/${jar.baseName}:${project.version}"
+    tags.add("${project.dockerTagBase}/${jar.baseName}:${project.version}".toString())
+    tags.add("${project.dockerTagBase}/${jar.baseName}:latest".toString())
 }
```
Comment (wtritch) [inline build.gradle]: This preserves the old tagging of images with snapshot versions (`:5.0.2-SNAPSHOT`) as well as tags the latest image with `:latest`

File: build.gradle [hunk 3]
```diff
@@ -244,14 +244,20 @@ task dockerCopyFiles(type: Copy) {
...
 
 task dockerPushImage(type: DockerPushImage) {
     dependsOn 'dockerBuildImage'
     imageName = "${project.dockerTagBase}/${jar.baseName}:${project.version}"
 }
 
+task dockerPushLatest(type: DockerPushImage) {
+    dependsOn 'dockerBuildImage'
+    imageName = "${project.dockerTagBase}/${jar.baseName}:latest"
+}
```
Comment (wtritch) [inline build.gradle]: This pushes the `:latest` image to docker-hub.  Ideally our CI process would do this IFF the branch name is `develop`  Something like a command-line build step that conditionally runs `./gradlew dockerPushLatest`

File: src/main/resources/application.yml [hunk 4]
```diff
@@ -20,10 +20,79 @@ management:
 api:
   context-path: "/api"
 
+security:
+  basic:
+    enabled: \${SECURITY_ENABLED:true}
+  user:
+    name: "admin"
+    password: "admin"
```
Comment (wtritch) [inline src/main/resources/application.yml]: I did have a question about the `security.user.name` and `.password` properties.  By moving the configuration into the embedded application.yml, those WILL be the values in any environment unless overridden.  Are those values overridden in SBAC's production/staging environments?  I believe the default is to pick a random alphanumeric generated at runtime.

File: src/main/resources/application.yml [hunk 5]
```diff
@@ -20,10 +20,79 @@ management:
 api:
   context-path: "/api"
 
+security:
+  basic:
+    enabled: \${SECURITY_ENABLED:true}
```
Comment (esotrope) [inline src/main/resources/application.yml]: what's the `\` for?

Comment (wtritch) [inline src/main/resources/application.yml]: The escaping slashes on Spring property references are due to our use of Gradle's "expand" function which complains if it finds an unknown property reference.

An old spring boot commit explains that Spring made this change expressly for Gradle support:
https://github.com/spring-projects/spring-boot/commit/c0c67f2593dbfd17aa304b43f4da3a3678fa58eb

Comment (wtritch) [reply src/main/resources/application.yml]: Explained in another comment =)

Comment (esotrope) [reply src/main/resources/application.yml]: 👍 gracias

File: build.gradle (hunk 2 above)
Comment (millerts) [reply build.gradle]: Is there something that uses the snapshot docker images?  @bdrainer , if this isn't used can Will just remove it?

File: src/main/resources/application.yml (hunk 4 above)
Comment (millerts) [reply src/main/resources/application.yml]: I think we'll need to pull this out or update the deployment files for all environments

Comment (wtritch) [reply src/main/resources/application.yml]: Agreed, removed.  If somebody wants to set it locally, they can do so via runtime environment variables, java properties, docker-compose environment variables, etc, etc

**Pull request merged**